File: SearchSample/Objective-C/SearchSample/KeywordViewController.m
//
//  KeywordViewController.m
//  SearchSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

#import "KeywordViewController.h"


@implementation KeywordViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([segue.identifier isEqualToString:@"keywordtoResultSegue"]) {
        ResultViewController *resultViewController = segue.destinationViewController;
        [resultViewController searchByKeyword:sender];

    }
}

- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
    if (searchText.length > 0) {
        NTAutocompleteParameter *param = [[NTAutocompleteParameter alloc] initWithKeyword:searchText];
        
        [NTAutocompleteService executeAsync:param Completion:^(NTAutocompleteResultSet * resultSet, NSError *error) {
            if (!error) {
                _keywords = resultSet.results;
            }
            
            dispatch_async(dispatch_get_main_queue(), ^{
                [_tableView reloadData];
            });
        }];
    }
}

- (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar {
    [searchBar resignFirstResponder];
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    NTAutocompleteResult *keyword = _keywords[indexPath.row];
    [self performSegueWithIdentifier:@"keywordtoResultSegue" sender:keyword.name];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    NTAutocompleteResult *keyword = _keywords[indexPath.row];
    cell.textLabel.text = keyword.name;
    
    return cell;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _keywords != nil ? _keywords.count : 0;
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}


@end

File: SearchSample/Swift/SearchSample/KeywordViewController.swift
//
//  KeywordViewcontroller.swift
//  SearchSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

import UIKit
import NOSTRASDK

class KeywordViewController: UIViewController, UISearchBarDelegate, UITableViewDelegate, UITableViewDataSource {

    
    @IBOutlet weak var tableView: UITableView!
    
    @IBOutlet weak var searchBar: UISearchBar!
    
    
    var keywords: [NTAutocompleteResult]?;
    
    var param: NTAutocompleteParameter?;
    
    override func viewDidLoad() {
        super.viewDidLoad();
    }
    
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "keywordtoResultSegue" {
            let resultViewController = segue.destination as! ResultViewController;
            
            resultViewController.searchByKeyword(sender as? String);
        }
    }
    
    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
        
        if searchText.characters.count > 0 {
            param = NTAutocompleteParameter(keyword: searchText);
            
            NTAutocompleteService.executeAsync(param!) { (resultSet, error) in
                
                DispatchQueue.main.async(execute: {
                    if error == nil {
                        self.keywords = resultSet?.results;
                    }
                    else {
                        self.keywords = [];
                        print("error: \(error?.description)");
                    }
                    self.tableView.reloadData();
                })
                
            }
        }
        
        
        
        
    }

    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
        searchBar.resignFirstResponder();
    }
    
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let keyword = keywords![indexPath.row];
        self.performSegue(withIdentifier: "keywordtoResultSegue", sender: keyword.name);
    }
    
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell");
        let keyword = keywords![indexPath.row];
        
        cell?.textLabel?.text = keyword.name;
        
        return cell!;
        
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return keywords != nil ? (keywords?.count)! : 0;
    }
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    
    
}
